src/main/java/calculator/ver3/Operation.java

```diff
@@ -0,0 +1,54 @@
+package calculator.ver3;
+
+public enum Operation {
+    PLUS("+") {
+        public MyNumberVer3 apply(MyNumberVer3 x, MyNumberVer3 y) {
+            return new MyNumberVer3(x.getNumber() + y.getNumber());
+        }
+    },
+    MINUS("-") {
+        public MyNumberVer3 apply(MyNumberVer3 x, MyNumberVer3 y) {
+            return new MyNumberVer3(x.getNumber() - y.getNumber());
+        }
+    },
+    MULTIPLE("*") {
+        public MyNumberVer3 apply(MyNumberVer3 x, MyNumberVer3 y) {
+            return new MyNumberVer3(x.getNumber() * y.getNumber());
+        }
+    },
+    DIVIDE("/") {
+        public MyNumberVer3 apply(MyNumberVer3 x, MyNumberVer3 y) {
+            valid(y);
+            return new MyNumberVer3(x.getNumber() / y.getNumber());
+        }
+    };
+
+    private final String symbol;
+
+    Operation(String symbol) {
+        this.symbol = symbol;
+    }
+
```

Comment:
```suggestion
        PLUS("+", (x, y) -> x + y),
        MINUS("-", (x, y) -> x - y),
        MULTIPLY("*", (x, y) -> x * y),
        DIVIDE("/", (x, y) -> x / y);

        private String symbol;
        private BiFunction<MyNumberVer3, MyNumberVer3, MyNumberVer3> expression;

        Operator(String operator, BiFunction<MyNumberVer3, MyNumberVer3, MyNumberVer3> expression) {
            this.symbol = operator;
            this.expression = expression;
        }
```

람다와 BiFunction을 이용하면 이런 형태로 간결하게 코드를 만들 수 있어요.
이 부분은 학습에 도움이 되셨으면 하는 바램에 드리는 의견이니 반영하지 않고 그렇구나 정도로 이해하고 넘어가셔도 괜찮습니다. 😄

Reply:
람다와 BiFunction 을 사용하면 이렇게 간결해지는군요!!
몰랐던 부분이였습니다 감사합니다